code/game/objects/machinery/doors/airlock.dm

```diff
@@ -198,6 +223,25 @@
 				overlays += image(icon, "panel_open")
 			if(welded)
 				overlays += image(icon, "welded")
+		if(hasPower() && unres_sides)
+			for(var/heading in list(NORTH,SOUTH,EAST,WEST))
+				if(!(unres_sides & heading))
+					continue
+				var/image/access_overlay = image('icons/obj/doors/overlays.dmi', "unres_[heading]", layer = ABOVE_ALL_MOB_LAYER, pixel_y = -4)
+				switch(heading)
+					if(NORTH)
+						access_overlay.pixel_x = 0
+						access_overlay.pixel_y = 32
+					if(SOUTH)
+						access_overlay.pixel_x = 0
+						access_overlay.pixel_y = -32
+					if(EAST)
+						access_overlay.pixel_x = 32
+						access_overlay.pixel_y = 0
+					if(WEST)
+						access_overlay.pixel_x = -32
+						access_overlay.pixel_y = 0
+				overlays += access_overlay
```

Comment:
this stuff needs to be in update_overlays (and the stuff above for overlays)

Reply:
Funny thing this, it originally did live in update_overlays(), but update_icon() manages all door overlays and cuts everything else, so instead of refactoring the logic to use update_overlays() for everything I just moved all this stuff into update_icon.